setup/triggers/amazon.md

```diff
@@ -0,0 +1,153 @@
+
+You have now set up an SNS topic and created a SQS queue which is subscribed to the topic. When messages are delivered to the topic, they will be delivered to the queue.
+
+## Configuring an S3 bucket to send notifications
+
+The next step is to create an S3 bucket, or use an existing bucket, and automatically deliver notifications when a file is uploaded. The following commands can be used to create a bucket and configure the notifications.
+
+
+* Now you have to tell AWS that your SNS topic can receive messages from your S3 bucket. To do this, you'll need to change the Access Policy for the SNS topic. You can do this by issuing the following command, swapping out the values with the resources created above:
+
+```
+
+bash4.4$ aws sns set-topic-attributes --topic-arn <YOUR_TOPIC_ARN_HERE> --attribute-name Policy --attribute-value "{\"Version\":\"2008-10-17\",\"Id\":\"__default_policy_ID\",\"Statement\":[{\"Sid\":\"s3Publish\",\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"*\"},\"Action\":[\"SNS:GetTopicAttributes\",\"SNS:Publish\"],\"Resource\":\"arn:aws:sns:us-east-2:791719295754:test-topic\",\"Condition\":{\"ArnLike\":{\"aws:SourceArn\":\"arn:aws:s3:::spin-pubsub-test-bucket\"}}}]}"
+bash4.4$
+```
+* Now that you've given your topic permission to have messages published to it, you can update your bucket's `put-bucket-notification-configuration` with the following `spin-pubsub-notification.json`. You'll need to swap out any values for the values for your configuration.  
+
+```
+ bash4.4$ aws s3 mb s3://spin-pubsub-test-bucket/
+ make_bucket: spin-pubsub-test-bucket
+ # You need to create a file with the notification information in json format
+ bash4.4$ cat << EOF > /tmp/spin-pubsub-notification.json
+ {
+    "TopicConfigurations": [
+        {
+            "TopicArn": "arn:aws:sns:us-east-1:1234567890:aws-pubsub-test-topic",
+            "Events": [
+                "s3:ObjectCreated:*"
+            ]
+        }
+    ]
+ }
+EOF
+ bash4.4$ aws s3api put-bucket-notification-configuration --bucket spin-pubsub-test-bucket \
+ --notification-configuration file:///tmp/spin-pubsub-notification.json    
+ bash4.4$
+```
+* Congratulations! You have now done all of the configuration in AWS to send pubsub messages. All we have to do now is configure Spinnaker to subscribe to and retrieve these messages.
+
+## Writing your Echo configuration
+
+Amazon pubsub configuration is not yet supported via Halyard, so you'll need to define the details regarding your subscription in an echo-local.yml file. Put this echo-local.yml file in `~/.hal/<Deployment Profile>/profiles/` directory and Halyard will automatically deploy it alongside echo the next time you run `hal deploy apply`. Halyard's default profile is named `default`, so if you haven't set up a different profile, this `echo-local.yaml` file would go in the `~/.hal/default/profiles/` directory on your halyard instance.
```

Comment:
Can we note that PRs for halyard support would be a decent place to start for someone looking to help out with the code?

Reply:
Done.